pandas/core/frame.py

```diff
@@ -5008,7 +5008,7 @@ def set_axis(
         labels,
         *,
         axis: Axis = 0,
-        copy: bool = True,
+        copy: bool | None = None,
```

Comment:
you'll need to update the annotation and default value for `Series.set_axis` as well